aws/rust-runtime/aws-config/src/environment/timeout_config.rs

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Load timeout configuration properties from an AWS profile
+
+use aws_smithy_types::timeout::{TimeoutConfigBuilder, TimeoutConfigError};
+use aws_types::os_shim_internal::Env;
+use std::time::Duration;
+
+const ENV_VAR_CONNECT_TIMEOUT: &str = "AWS_CONNECT_TIMEOUT";
+const ENV_VAR_TLS_NEGOTIATION_TIMEOUT: &str = "AWS_TLS_NEGOTIATION_TIMEOUT";
+const ENV_VAR_READ_TIMEOUT: &str = "AWS_READ_TIMEOUT";
+const ENV_VAR_API_CALL_ATTEMPT_TIMEOUT: &str = "AWS_API_CALL_ATTEMPT_TIMEOUT";
+const ENV_VAR_API_CALL_TIMEOUT: &str = "AWS_API_CALL_TIMEOUT";
+
+/// Load a timeout_config from environment variables
+///
+/// This provider will check the values of the following variables in order to build a `TimeoutConfig`
```

Comment:
Add a note in here indicating the units these environment variables are in, and that decimals are accepted. Looks like it's seconds?

Reply:
see above